### PR TITLE
Add constraints to properties

### PR DESCRIPTION
**Pull request by marians**
### What this PR does / why we need it

To avoid simpe input mistakes, this PR adds constraints to numeric and string properties.

### Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: helm/cluster-aws/values.schema.json [hunk 1]
```diff
@@ -630,10 +643,15 @@
                 },
                 "name": {
                     "description": "Unique identifier, cannot be changed after creation.",
+                    "maxLength": 10,
```
Comment (AndiDog) [inline helm/cluster-aws/values.schema.json:646]: I also saw these numbers for vintage clusters. Do we _really_ need to restrict at 10 characters? If not, I'd rather leave this at Kubernetes' maximum name limit. Same for the minimum.

File: helm/cluster-aws/values.schema.json [hunk 2]
```diff
@@ -658,6 +676,9 @@
             "properties": {
                 "ami": {
                     "description": "If specified, this image will be used to provision EC2 instances.",
+                    "maxLength": 100,
+                    "minLength": 8,
+                    "pattern": "^ami-[0-9abzdefg]+$",
```
Comment (AndiDog) [inline helm/cluster-aws/values.schema.json:681]: Where's this regex from? Do they only use this alphabet choice?

Comment (marians) [reply helm/cluster-aws/values.schema.json:681]: I looked at documentation and all examples I found used a hexadecimal ID with the `ami-` prefix prepended.

**Comment by fiunchinho** [comment]
What's the state of this PR @marians ? I see that it's still a draft.

File: helm/cluster-aws/values.schema.json (hunk 2 above)
Comment (marians) [inline helm/cluster-aws/values.schema.json:681]:
```suggestion
                    "pattern": "^ami-[0-9abcdefg]+$",
```

**Comment by marians** [comment]
I have parked this PR since I didn't find the time to work out which constraints would actually make sense. But I think it's still needed, as we have many strings where e. g. adding a blank space by mistake could break the whole thing and this would be easy to catch.

**Comment by AndiDog** [comment]
Let's close this since the diff is outdated. We'd have to redo the change once time allows and if we still find it useful.

**Pull request closed without merging**